spotmarketrequest.go

```diff
@@ -0,0 +1,108 @@
+package packngo
+
+import (
+	"fmt"
+	"math"
+)
+
+const spotMarketRequestBasePath = "/spot-market-requests"
+
+type SpotMarketRequestService interface {
+	List(string) ([]SpotMarketRequest, *Response, error)
+	Create(*SpotMarketRequestCreateRequest, string) (*SpotMarketRequest, *Response, error)
+	Delete(string, bool) (*Response, error)
+	Get(string) (*SpotMarketRequest, *Response, error)
+}
+
+type SpotMarketRequestCreateRequest struct {
+	DevicesMax  int        `json:"devices_max"`
+	DevicesMin  int        `json:"devices_min"`
+	EndAt       *Timestamp `json:"end_at,omitempty"`
+	FacilityIDs []string   `json:"facility_ids"`
+	MaxBidPrice float64    `json:"max_bid_price"`
+
+	Parameters InstanceParameters `json:"instance_parameters"`
+}
+
+type SpotMarketRequest struct {
+	SpotMarketRequestCreateRequest
+	ID         string     `json:"id"`
+	Devices    []Device   `json:"devices"`
+	Facilities []Facility `json:"facilities"`
+	Project    Project    `json:"project"`
+	Href       string     `json:"href"`
+}
+
+type InstanceParameters struct {
```

Comment:
Perhaps  `RequestInstanceParameters`?

Reply:
fixed to SpotMarketRequestInstanceParameters